drivers/rpmsg/rpmsg_core.c

```diff
@@ -85,6 +85,134 @@ void rpmsg_destroy_ept(struct rpmsg_endpoint *ept)
 }
 EXPORT_SYMBOL(rpmsg_destroy_ept);
 
+/**
```

Comment:
you need to watch out for the software license here. The OpenAmp code is licensed as BSD3 and you are using it in a GPLv2 file. Have you checked with legal that this is fine?

Reply:
The concept is from OpenAmp, but the code is written from scratch, is it still a legal issue?

Reply:
if you did reimplement you should add a comment to avoid confusing the reader.

I will not provide any legal advice or statement, just highlighting the fact that when reusing concepts you need to look into license opens, e.g. run this code through accepted tools to avoid GPL violation issues.